src/components/VHomeGallery/VHomeGallery.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+  <!-- Wrapper element to center the grid if space is more than 5 columns. -->
+  <div ref="el" class="mx-10 flex flex-row items-center justify-end me-12">
+    <!-- Image grid only occupies as much width as needed. -->
+    <div
+      class="home-gallery inline-grid grid-flow-col grid-rows-3 gap-8"
+      :style="{
+        gap: `${space}px`,
+        gridTemplateColumns: `repeat(${columnCount}, minmax(0, 1fr))`,
+      }"
+    >
+      <ClientOnly>
+        <Component
+          :is="prefersReducedMotion ? 'div' : 'Transition'"
+          v-for="(image, idx) in imageList"
+          :key="idx"
+          name="fade"
+          mode="out-in"
+          appear
+        >
+          <VLink
+            class="home-cell rounded-full p-1 focus:bg-white"
+            :class="idx >= imageCount ? 'hidden' : 'block'"
+            :style="{ '--transition-delay': `${idx * 0.05}s` }"
+            :href="image.url"
+          >
+            <img
+              :height="dimens"
+              :width="dimens"
+              :src="image.src"
+              :alt="image.title"
+              :title="image.title"
+            />
+          </VLink>
+        </Component>
+      </ClientOnly>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  useContext,
+  useRouter,
+  PropType,
+  ref,
+} from "@nuxtjs/composition-api"
+
+import { useReducedMotion } from "~/composables/use-media-query"
+import useResizeObserver from "~/composables/use-resize-observer"
+
+import imageInfo from "~/assets/homepage_images/image_info.json"
+
+export const GALLERY_SETS = [
+  "universe",
+  "pottery",
+  "olympics",
+  "random",
+] as const
+export type GallerySet = typeof GALLERY_SETS[number]
+
+/**
+ * Displays a grid of images for the homepage, with each image linking to its
+ * single result page. The number of columns automatically adjusts to the width
+ * of the container upto a max of 5.
+ */
+export default defineComponent({
+  name: "VHomeGallery",
+  props: {
+    /**
+     * the set of images to use for the gallery grid
+     */
+    set: {
+      type: String as PropType<GallerySet>,
+      required: false,
+      default: "random",
+      validator: (val: GallerySet) => GALLERY_SETS.includes(val),
+    },
+  },
+  setup(props) {
+    const { app } = useContext()
+    const router = useRouter()
+    const prefersReducedMotion = useReducedMotion()
+
+    const dimens = 152 // px
+    const space = 24 // px; 32px space - 4px padding on both sides
+
+    const rowCount = 3
+    const columnCount = computed(() => {
+      console.log(gridDimens.value.width)
```

Comment:
```suggestion
```

Reply:
Left-over logging

Reply:
Ah I always forget! I should add a lint rule for this. Removed in 302f703.

Reply:
I do too :D